geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/vector/ArrowAttributeWriter.scala

```diff
@@ -104,229 +85,247 @@ object ArrowAttributeWriter {
+          case ObjectType.MAP      => new ArrowMapWriter(name, bindings(1), bindings(2), encoding, metadata, factory)
+          case ObjectType.BYTES    => new ArrowBytesWriter(name, metadata, factory)
+          case ObjectType.UUID     => new ArrowStringWriter(name, metadata, factory)
           case _ => throw new IllegalArgumentException(s"Unexpected object type ${bindings.head}")
         }
+    }
+  }
 
-      case Some(dict) =>
-        val dictionaryEncoding = dict.encoding
-        val dictionaryType = TypeBindings(bindings, encoding)
-        if (dictionaryEncoding.getIndexType.getBitWidth == 8) {
-          val vector = toVector[TinyIntVector](MinorType.TINYINT.getType, dictionaryEncoding, metadata)
-          new ArrowDictionaryByteWriter(vector, dict, dictionaryType)
-        } else if (dictionaryEncoding.getIndexType.getBitWidth == 16) {
-          val vector = toVector[SmallIntVector](MinorType.SMALLINT.getType, dictionaryEncoding, metadata)
-          new ArrowDictionaryShortWriter(vector, dict, dictionaryType)
-        } else {
-          val vector = toVector[IntVector](MinorType.INT.getType, dictionaryEncoding, metadata)
-          new ArrowDictionaryIntWriter(vector, dict, dictionaryType)
-        }
+  private def dictionary(
+      name: String,
+      dictionary: ArrowDictionary,
+      metadata: Map[String, String],
+      factory: VectorFactory): ArrowAttributeWriter = {
+    dictionary.encoding.getIndexType.getBitWidth match {
+      case 8  => new ArrowDictionaryByteWriter(name, dictionary, metadata, factory)
+      case 16 => new ArrowDictionaryShortWriter(name, dictionary, metadata, factory)
+      case 32 => new ArrowDictionaryIntWriter(name, dictionary, metadata, factory)
+      case w  => throw new IllegalArgumentException(s"Unsupported dictionary encoding width: $w")
+    }
+  }
+
+  private def date(
+      name: String,
+      encoding: Encoding,
+      metadata: Map[String, String],
+      factory: VectorFactory): ArrowAttributeWriter = {
+    encoding match {
+      case Encoding.Min => new ArrowDateSecondsWriter(name, metadata, factory)
+      case Encoding.Max => new ArrowDateMillisWriter(name, metadata, factory)
+    }
+  }
+
+  private def geometry(
+      name: String,
+      binding: ObjectType,
+      encoding: Encoding,
+      metadata: Map[String, String],
+      factory: VectorFactory): ArrowGeometryWriter = {
+    val m = metadata.asJava
+    val vector = (binding, encoding, factory) match {
+      case (ObjectType.POINT, Encoding.Min, FromStruct(c))              => new PointFloatVector(name, c, m)
+      case (ObjectType.POINT, Encoding.Min, FromAllocator(c))           => new PointFloatVector(name, c, m)
+      case (ObjectType.POINT, Encoding.Max, FromStruct(c))              => new PointVector(name, c, m)
+      case (ObjectType.POINT, Encoding.Max, FromAllocator(c))           => new PointVector(name, c, m)
```

Comment:
Can the match be
case (ObjectType.POINT, Encoding.Max, _)
to save a few lines?

Reply:
i don't think so... there are 2 construtors, one takes a vector and the second takes an allocator. so the match here is selecting the right constructor. (not very apparent when you're just looking at it)